catalog-rest-service/src/main/resources/json/schema/tests/basic.json

```diff
@@ -0,0 +1,39 @@
+{
+  "$id": "https://open-metadata.org/schema/test/basic.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Basic",
+  "description": "This schema defines basic types that are used by other test schemas.",
+  "definitions": {
+    "testType": {
+      "javaType": "org.openmetadata.catalog.type.TestType",
+      "description": "This schema defines the type used for describing different types of tests.",
+      "type": "string",
+      "enum": ["TableTest", "MultiTableTest", "ColumnTest"],
+      "javaEnums": [
+        {
+          "name": "TableTest"
+        },
+        {
+          "name": "MultiTableTest"
+        },
+        {
+          "name": "ColumnTest"
+        }
+      ]
+    },
```

Comment:
If we add tests for say `reports`, `dashboards` and other data assets, would they be defined here?

Reply:
Yes thats the idea . One thing with the JsonSchema lets say we define type like above but I want to designate a type in the test definition. i.e call out a value in the enum to be the type. I am not able to find a way to configure that.
```
{
  "$id": "https://open-metadata.org/schema/tests/column/columnValuesToMatchRegex.json",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "title": "columnValuesToBeUnique",
  "description": "This schema defines the test ColumnValuesToMatchRegex. Test the values in a column to match a given regular expression. ",
  "type": "object",
  "javaType": "org.openmetadata.catalog.entity.tests.column.ColumnValuesToMatchRegex",
  "properties": {
    "column": {
      "description": "Name of the column in a table.",
      "type": "string"
    },
    "regex": {
      "description": "The regular expression the column entries should match.",
      "type": "string"
    }
   "type":  {
        "ref":"./basic.json#definitions/testType#ColumnTest"
      }
  },
  "required": ["column", "regex"]
}